src/bot.js

```diff
@@ -51,7 +48,10 @@ class Bot {
     // give informations about env vars
     this.checkEnvVars();
     this.id = process.env.BOTFUEL_APP_TOKEN;
-    this.adapter = this.getAdapter(this.config.adapter);
+
+    this.adapterManager = new AdapterManager(this);
+    this.adapter = this.adapterManager.resolve(this.config.adapter);
```

Comment:
We don't need the adapterManager except for resolving the adapter so I would have written `this.adapter = new AdapterManager(this).resolve()` or made the method static and use `this.adapter = AdapterManager.resolve(config)`